streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -228,12 +228,16 @@ public Subscription subscription(Set<String> topics) {
         // 2. within each client, tasks are assigned to consumer clients in round-robin manner.
         Map<UUID, Set<String>> consumersByClient = new HashMap<>();
         Map<UUID, ClientState<TaskId>> states = new HashMap<>();
-
+        Set<String> regexSubscribedTopics = new HashSet<>();
         // decode subscription info
         for (Map.Entry<String, Subscription> entry : subscriptions.entrySet()) {
             String consumerId = entry.getKey();
             Subscription subscription = entry.getValue();
 
+            if (streamThread.builder.sourceTopicPattern() != null) {
+                regexSubscribedTopics.addAll(subscription.topics());
```

Comment:
I think we should replace with the subscription list here as well, instead of incrementally add to it, since for regex subscription some originally matched topics could be deleted from the broker and hence gets removed from the subscription list.

In fact, in this case, do we still need `regexSubscribedTopics`, or could we just do ` subscriptionUpdates.updateTopics(subscription.topics())`?